core/test_network/src/lib.rs

```diff
@@ -832,29 +836,25 @@ impl TestConfiguration for Configuration {
 }
 
 impl TestClientConfiguration for ClientConfiguration {
-    fn test(api_url: &str, telemetry_url: &str) -> Self {
+    fn test(api_url: &SocketAddr, telemetry_url: &SocketAddr) -> Self {
         let mut configuration = iroha_client::samples::get_client_config(&get_key_pair());
-        configuration.torii_api_url = if api_url.starts_with("http") {
-            small::SmallStr::from_str(api_url)
-        } else {
-            small::SmallStr::from_str(&("http://".to_owned() + api_url))
-        };
-        configuration.torii_telemetry_url = if telemetry_url.starts_with("http") {
-            small::SmallStr::from_str(telemetry_url)
-        } else {
-            small::SmallStr::from_str(&("http://".to_owned() + telemetry_url))
-        };
+        configuration.torii_api_url = format!("http://{api_url}")
```

Comment:
We're testing the parsing? Shouldn't this be handled by a unit test inside `primitives` and all other instances, handled by a purpose built-macro that does some checks?

Reply:
In client config `api_url` is an actual URL, not a socket address, and, bizzarely, there's no reasonable way to construct `url::Url` other that parsing a string.

Reply:
Interesting. I'd imagine that a macro like `url!(http://{api_url}:8080/post_tx) would be tough to write, but if it expands to something along the lines of 
```rust
Url { 
   protocol: url::Http(), 
   host: api_url.clone(), 
   port: Some(8080), 
   segments: vec!["post_tx"], 
}
```
Could do the trick.

Reply:
Resolving to be implemented as part of https://github.com/hyperledger/iroha/issues/3385